### PR TITLE
[script.myepisodecalendar] 1.2.1

### PR DESCRIPTION
**Pull request by Connum**
### Description
My AddOn is already in the repos somewhere (submitted via the mailing list before the switch to Git), this is an update to the latest version.

new in v1.2.1:
+ Optimized the detection of metadata (TV show, season, episode)
+ Only run if detected media type is "tvshow", "season" or "episode" (or unset):
   This prevents the "XY not found" notification when watching movies.

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by Razzeee** [comment]
It's in the isengard branch.

**Comment by Connum** [comment]
Good to know, thanks! As I didn't test the new version in Isengard though, I think it's fine if the updated version makes it into Jarvis and up only.

File: script.myepisodecalendar/main.py [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+class Monitor(xbmc.Monitor):
+    def __init__( self, *args, **kwargs ):
+        xbmc.Monitor.__init__( self )
+        self.action = kwargs['action']
+
+    def onSettingsChanged( self ):
+        log('#DEBUG# onSettingsChanged')
+        # remove shows added by plugin when integration is disabled
+        if __nextAired__ and __addon__.getSetting('tvsna-enabled') == 'false':
+            __nextAired__.setSetting("ExtraShows", re.sub(r"\D?mec\d+", '', __nextAired__.getSetting("ExtraShows")))
```
Comment (romanvm) [inline script.myepisodecalendar/main.py]: Probable an oversight by the previous reviewer but normally we do not allow addons to manipulate settings of other addons.

File: script.myepisodecalendar/main.py [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+                log("showId %s is in shows list" % cleanId)
+                CleanedExtraShows.append('mec' + cleanId)
+            else:
+                log("remove %s from shows list" % cleanId)
+                removedCount = removedCount+1
+
+        ExtraShows = CleanedExtraShows
+
+    # save new ExtraShows list to script.tv.show.next.aired settings
+    __nextAired__.setSetting('ExtraShows', ','.join(ExtraShows))
```
Comment (romanvm) [inline script.myepisodecalendar/main.py]: The same as above

File: script.myepisodecalendar/program.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import xbmcaddon
+
+try:
+    startparam = sys.argv[1]
+except:
+    startparam = False
+
+if (startparam is False):
```
[the remark this answers is not in the document]
Comment (Connum) [reply script.myepisodecalendar/program.py]: That's a good question... My addon is based on script.myepisodes and this was probably already in. This addon was also the first time that I worked with Python, so I didn't have any idea about the language-specific quirks and odds... I'll make sure to address this!

File: script.myepisodecalendar/main.py [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+__addon__         = xbmcaddon.Addon()
+__addonid__       = __addon__.getAddonInfo('id')
+__cwd__           = __addon__.getAddonInfo('path')
+__icon__          = __addon__.getAddonInfo("icon")
+__scriptname__    = __addon__.getAddonInfo('name')
+__version__       = __addon__.getAddonInfo('version')
+__language__      = __addon__.getLocalizedString
+__resource_path__ = os.path.join(__cwd__, 'resources', 'lib')
+__resource__      = xbmc.translatePath(__resource_path__).decode('utf-8')
+__datapath__      = os.path.join(xbmc.translatePath('special://masterprofile/addon_data/').decode('utf-8'), __addonid__)
```
[the remark this answers is not in the document]
Comment (Connum) [reply script.myepisodecalendar/main.py]: And again, that's a leftover from script.myepisodes - so I guess I'll have to rename all the variables then.

File: script.myepisodecalendar/main.py (hunk 1 above)
Comment (Connum) [reply script.myepisodecalendar/main.py]: Ok, I see... So I'll have to remove the next aired functionality completely, that's a bummer.

File: script.myepisodecalendar/resources/lib/myepisodecalendar.py [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+        seen_url = "%s/actions/watchList.php" % MYEPISODE_URL
+        data = self.send_req(seen_url, episode_add_data)
+
+        if data is None:
+            return False
+
+        return True
+
+    def getTVDBIDFromShowTitle(self, title, lang='en'):
+        GetSeriesURL = 'http://www.thetvdb.com/api/GetSeries.php?seriesname=%s&language=%s' % (urllib.quote(title), lang)
```
[the remark this answers is not in the document]
Comment (Connum) [reply script.myepisodecalendar/resources/lib/myepisodecalendar.py]: Also a leftover - and honestly, I also didn't know about that.

**Comment by Connum** [comment]
Ok, so that was a lot to process... Solving all of this makes it a major rewrite, so I suppose it's best to close this PR, get everything solved and add a new one for a new version.

File: script.myepisodecalendar/main.py (hunk 1 above)
Comment (Razzeee) [reply script.myepisodecalendar/main.py]: Can you explain what this tries to accomplish?

Comment (Connum) [reply script.myepisodecalendar/main.py]: The integration, when enabled,  adds all your shows from myepisodecalendar.com to the list of shows of the TV Show Next Aired addon. And this specific part of the code removes those added shows again after saving the settings and having disabled the integration.

**Comment by learningit** [comment]
Yes it's probably best to open a new PR when you get everything fixed.
I'm going to close this for now.

**Pull request closed without merging**